packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1126,6 +1127,21 @@ class _NestedScrollCoordinator implements ScrollActivityDelegate, ScrollHoldCont
     _outerPosition?.setParent(_parent ?? PrimaryScrollController.maybeOf(_state.context));
   }
 
+  bool shuldIgnoreEnsureVisible(
+    _NestedScrollPosition target, {
+    ScrollableState? previousScrollable,
+  }) {
+    if (target == _outerPosition &&
+        previousScrollable != null &&
+        _innerPositions.contains(previousScrollable.position)) {
+      // If the target is the outer position and the previous scrollable is an inner position,
+      // we should ignore the ensureVisible call. This is because the inner position will
+      // already have been scrolled to make the target visible.
+      return true;
+    }
+    return false;
```

Comment:
```suggestion
      // If the target is the outer position and the previous scrollable is an inner position,
      // we should ignore the ensureVisible call. This is because the inner position will
      // already have been scrolled to make the target visible.
    return target == _outerPosition &&
        previousScrollable != null &&
        _innerPositions.contains(previousScrollable.position);
```